### PR TITLE
add model optimization

### PR DESCRIPTION
**Pull request by fsx950223**
add model optimization example
Screenshot from 2020-10-21 18-53-37
model prune

### CODE REVIEW

**Comment by fsx950223** [comment]
I could get 72 mAP0.5 when I train the model with Estimator but I only get 17mAP0.5 when I train the model with Keras on VOC2007. Do you have any idea? @mingxingtan

File: efficientdet/keras/train_lib.py [hunk 1]
```diff
@@ -174,7 +289,7 @@ def get_optimizer(params):
   if params['optimizer'].lower() == 'sgd':
     logging.info('Use SGD optimizer')
     optimizer = tf.keras.optimizers.SGD(
-        learning_rate, momentum=params['momentum'])
+        learning_rate, momentum=params['momentum'], nesterov=True)
```
Comment (mingxingtan) [inline efficientdet/keras/train_lib.py:292]: do you get better results with nesterov?

File: efficientdet/keras/train_lib.py [hunk 2]
```diff
@@ -456,9 +584,15 @@ class and box losses from all levels.
               cls_loss, [bs, width, height, -1, self.config.num_classes])
         cls_loss *= tf.cast(
             tf.expand_dims(
-                tf.not_equal(labels['cls_targets_%d' % (level + 3)], -2), -1),
+                tf.not_equal(
+                    labels['cls_targets_%d' % (level + self.config.min_level)],
```
Comment (mingxingtan) [inline efficientdet/keras/train_lib.py:588]: Good catch!

**Review by mingxingtan — APPROVED** [review]
Very cool feature. Thanks @fsx950223

File: efficientdet/keras/train_lib.py (hunk 1 above)
Comment (fsx950223) [reply efficientdet/keras/train_lib.py:292]: I removed nesterov and got 39 mAP0.5

**Pull request merged**